source/upgrading/upgrade/upgrade-4.13.rst

```diff
@@ -0,0 +1,360 @@
+
+#. If you are running a usage server or usage servers, stop those as well:
+
+   .. parsed-literal::
+
+      $ sudo service cloudstack-usage stop
+
+#. Make a backup of your MySQL database. If you run into any issues or
+   need to roll back the upgrade, this will assist in debugging or
+   restoring your existing environment. You'll be prompted for your
+   password.
+
+   .. parsed-literal::
+
+      $ mysqldump -u root -p cloud > cloud-backup_`date '+%Y-%m-%d'`.sql
+      $ mysqldump -u root -p cloud_usage > cloud_usage-backup_`date '+%Y-%m-%d'`.sql
+
+#. **(KVM Only)** If primary storage of type local storage is in use, the
+   path for this storage needs to be verified to ensure it passes new
+   validation. Check local storage by querying the cloud.storage\_pool
+   table:
+
+   .. parsed-literal::
+
+      $ mysql -u cloud -p -e "select id,name,path from cloud.storage_pool where pool_type='Filesystem'"
+
+   If local storage paths are found to have a trailing forward slash,
+   remove it:
+
+   .. parsed-literal::
+
+      $ mysql -u cloud -p -e 'update cloud.storage_pool set path="/var/lib/libvirt/images" where path="/var/lib/libvirt/images/"';
+
+
+.. _ubuntu412:
+.. _apt-repo412:
+
+Management Server
+-----------------
+
+Ubuntu
+######
+
+If you are using Ubuntu, follow this procedure to upgrade your packages. If
+not, skip to step :ref:`rhel411`.
+
+.. note::
+   **Community Packages:** This section assumes you're using the community
+   supplied packages for CloudStack. If you've created your own packages and
+   APT repository, substitute your own URL for the ones used in these examples.
+
+The first order of business will be to change the sources list for
+each system with CloudStack packages. This means all management
+servers, and any hosts that have the KVM agent. (No changes should
+be necessary for hosts that are running VMware or Xen.)
+
+
+
+Start by opening ``/etc/apt/sources.list.d/cloudstack.list`` on
+any systems that have CloudStack packages installed.
+
+This file should have one line, which contains:
+
+.. parsed-literal::
+
+   deb http://download.cloudstack.org/ubuntu precise 4.12
+
+We'll change it to point to the new package repository:
+
+.. parsed-literal::
+
+   deb http://download.cloudstack.org/ubuntu precise |version|
+
+Setup the public key for the above repository:
+
+.. parsed-literal::
+
+   wget -qO - http://download.cloudstack.org/release.asc | sudo apt-key add -
+
+If you're using your own package repository, change this line to
+read as appropriate for your |version| repository.
+
+#. Now update your apt package list:
+
+   .. parsed-literal::
+
+      $ sudo apt-get update
+
+#. Now that you have the repository configured, it's time to upgrade
+   the ``cloudstack-management`` package.
+
+   .. parsed-literal::
+
+      $ sudo apt-get upgrade cloudstack-management
+
+#. If you use CloudStack usage server
+
+   .. parsed-literal::
+
+      $ sudo apt-get upgrade cloudstack-usage
+
+
+.. _rhel412:
```

Comment:
rhel413
rpm-repo413

Reply:
I'm not sure how they're used, it's been from before I did not change it. Will need review/checking. It's also in the previous 4.11, 4.12 rst files. cc @PaulAngus

Reply:
as advised above, it's been like this for some time, and indeed needs updating

Reply:
fixed